replications/internal/queue_management_test.go

```diff
@@ -33,83 +29,39 @@ func TestCreateNewQueueDirExists(t *testing.T) {
 }
 
 func TestEnqueueScanLog(t *testing.T) {
-	t.Parallel()
-
-	// Initialize queue manager with zap observer (to allow assertions on log messages)
-	enginePath, err := os.MkdirTemp("", "engine")
-	require.NoError(t, err)
-	queuePath := filepath.Join(enginePath, "replicationq")
-
-	observedZapCore, observedLogs := observer.New(zap.InfoLevel)
-	observedLogger := zap.New(observedZapCore)
-
-	qm := NewDurableQueueManager(observedLogger, queuePath)
+	queuePath, qm := initQueueManager(t)
 	defer os.RemoveAll(filepath.Dir(queuePath))
 
 	// Create new queue
-	err = qm.InitializeQueue(id1, maxQueueSizeBytes)
+	err := qm.InitializeQueue(id1, maxQueueSizeBytes)
 	require.NoError(t, err)
 
 	// Enqueue some data
 	testData := "weather,location=us-midwest temperature=82 1465839830100400200"
+	qm.writeFunc = getTestWriteFunc(t, testData)
 	err = qm.EnqueueData(id1, []byte(testData))
 	require.NoError(t, err)
```

Comment:
Would it be practical to extend these tests to make sure that if `writeFunc` returns an error, the scanner does not get advanced?